dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/datavalue/hibernate/HibernateDataValueStore.java

```diff
@@ -267,6 +267,15 @@ public boolean dataValueExists(CategoryCombo combo) {
         .isEmpty();
   }
 
+  @Override
+  public boolean dataValueExistsForDataElement(String uid) {
+    return !getQuery("select 1 from DataValue dv where dv.dataElement.uid = :uid")
```

Comment:
Not sure it is needed since we use `select 1` but adding `limit 1` would make sure this query terminates as soon as 1 match is found.

Reply:
`setMaxResults(1)` is the HQL equivalent, which is used.